File: client-admin/src/app/shared/services/finance/account-replenishment.service.ts
import {Injectable} from '@angular/core';
import {AbstractService} from "../abstract.service";
import {HttpClient} from "@angular/common/http";
import {AccountReplenishment} from "../../models/finance/account-replenishment.model";

@Injectable({
  providedIn: 'root'
})
export class AccountReplenishmentService extends AbstractService<AccountReplenishment> {

  override resourceUrl = '/api/account-replenishments';

  constructor(public override httpClient: HttpClient) {
    super(httpClient);
  }
}

File: server/src/main/java/ua/com/alevel/service/finance/AccountReplenishmentService.java
package ua.com.alevel.service.finance;

import org.springframework.stereotype.Service;
import ua.com.alevel.persistence.crudhelper.CrudRepositoryHelper;
import ua.com.alevel.persistence.entity.finance.AccountReplenishment;
import ua.com.alevel.persistence.repository.finance.AccountReplenishmentRepository;
import ua.com.alevel.service.AbstractService;

@Service
public class AccountReplenishmentService extends AbstractService<AccountReplenishment, AccountReplenishmentRepository> {

    private final AccountReplenishmentRepository accountReplenishmentRepository;
    private final CrudRepositoryHelper<AccountReplenishment, AccountReplenishmentRepository> crudRepositoryHelper;

    public AccountReplenishmentService(AccountReplenishmentRepository accountReplenishmentRepository,
                                       CrudRepositoryHelper<AccountReplenishment, AccountReplenishmentRepository> crudRepositoryHelper) {
        super(accountReplenishmentRepository, crudRepositoryHelper);
        this.accountReplenishmentRepository = accountReplenishmentRepository;
        this.crudRepositoryHelper = crudRepositoryHelper;
    }
}
